frontend/class-frontend-page-type.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * @package WPSEO\Frontend
+ */
+
+/**
+ * Represents the classifier for determine the current opened page type.
+ */
+class WPSEO_Frontend_Page_Type {
+
+	/**
+	 * @var WPSEO_WooCommerce_Shop_Page
+	 */
+	protected $woocommerce_shop_page;
+
+	/**
+	 * Sets the WooCommmerce shop page object.
+	 *
+	 * @param WPSEO_WooCommerce_Shop_Page $woocommerce_shop_page The woocommerce shop page object.
+	 */
+	public function __construct( WPSEO_WooCommerce_Shop_Page $woocommerce_shop_page ) {
+		$this->woocommerce_shop_page = $woocommerce_shop_page;
+	}
+
+	/**
+	 * Checks if the current opened page is singular.
+	 *
+	 * @return bool True when singular.
+	 */
+	public function is_singular() {
+		if ( is_singular() ) {
+			return true;
+		}
+
+		if ( is_home() && 'page' === get_option( 'show_on_front' ) ) {
```

Comment:
I think that should be `'posts' === get_option( 'show_on_front' )`. You can look: https://github.com/WordPress/WordPress/blob/master/wp-includes/class-wp-query.php#L3812-L3821. If it's static page on frontend, then is_singular is already true.